cerk_port_mqtt_mosquitto/src/port_mqtt.rs

```diff
@@ -0,0 +1,229 @@
+                connection
+                    .client
+                    .subscribe(&subscribe_topic, connection.subscribe_qos.into())
+                    .unwrap();
+            }
+        });
+        callbacks.on_disconnect(|_, connection_id| {
+            debug!("{} disconnected: {}", id, connection_id);
+        });
+        connection.client.loop_until_disconnect(200).unwrap();
+    });
+
+    return Ok(sender);
+}
+
+fn send_cloud_event(
+    id: &InternalServerId,
+    event: &OutgoingCloudEvent,
+    connection: &Connection,
+    data: ArcData,
+) -> Result<()> {
+    let serialized = serde_json::to_string(&event.cloud_event)?;
+
+    if let Some(ref send_topic) = connection.send_topic {
+        let mut data_lock = data.lock().unwrap();
+        let message_id = connection.client.publish(
+            send_topic,
+            serialized.as_bytes(),
+            connection.send_qos.into(),
+            false,
+        )?;
+        data_lock
+            .unacked
+            .insert(message_id, event.routing_id.clone());
+        debug!("{} sent publish with id {}", id, message_id);
+    } else {
+        error!("{} not send_topic configured", id);
+    }
+    return Ok(());
+}
+
+/// This is the main function to start the port.
+pub fn port_mqtt_mosquitto_start(
+    id: InternalServerId,
+    inbox: BoxedReceiver,
+    sender_to_kernel: BoxedSender,
+) {
+    info!("start mqtt port with id {}", id);
+    let mut connection: Option<Connection> = None;
+    let mut sender: Option<Sender<ProcessingResult>> = None;
+    let data: ArcData = Arc::new(Mutex::new(Data {
+        unacked: HashMap::new(),
+    }));
+
+    loop {
+        match inbox.receive() {
+            BrokerEvent::Init => {
+                info!("{} initiated", id);
+            }
+            BrokerEvent::ConfigUpdated(config, _) => {
+                info!("{} received ConfigUpdated", &id);
+                match build_connection(&id, config) {
+                    Ok(new_connection) => {
+                        sender = Some(
+                            connect(
+                                id.clone(),
+                                new_connection.clone(),
+                                sender_to_kernel.clone_boxed(),
+                                data.clone(),
+                            )
+                            .unwrap(),
+                        );
+                        connection = Some(new_connection);
+                    }
+                    Err(e) => error!("failed to connect {:?}", e),
+                }
+            }
+            BrokerEvent::OutgoingCloudEvent(event) => {
+                debug!("{} cloudevent received", &id);
+                if let Some(ref connection) = connection {
+                    debug!("{} will send event out", &id);
+                    send_cloud_event(&id, &event, &connection, data.clone()).unwrap();
+                } else {
+                    error!("no active connection - cant send event");
```

Comment:
pleas send a `OutgoingCloudEventProcessed` with `ProcessingResult::TransientError` back if no connection could be established

Reply:
...and an acknowledgment is needed